content/concepts/introduction/protocols/multistream.md

```diff
@@ -1,7 +1,66 @@
 ---
 title: "Multistream Select"
-description: ""
+description: "Multistream Select is used to negotiate and upgrade the protocol of a connection between two peers in libp2p."
 weight: 26
 ---
 
-Coming soon!
+## Overview
+
+[Multistream Select](https://github.com/multiformats/multistream-select) is
+a protocol for friendly protocol multiplexing that allows for the negotiation
+of a [multicodec](https://github.com/multiformats/multicodec) between two entities.
+
+> Multicodecs are codecs that can handle multiple different encoding formats.
+> They are used to decode data and are self-describing, meaning they can identify
+> the data format they are decoding. This capability allows nodes to support multiple
+> protocols and efficiently switch between them depending on the task.
+
+Multistream is a general concept and refers to the ability
+to [multiplex multiple streams](../../multiplex/overview.md) of data over a single
+connection. This allows for multiple independent channels of communication to be
+established over a single connection rather than needing to open a separate
+connection for each channel.
+
+Multistream Select is a specific protocol built on top of the general
+concept of Multistream. It is used to negotiate and upgrade the protocol of a
+connection between two peers and to integrate new protocols as
+they become available. The actual protocol is fairly simple - it is a Multistream
+protocol with a Multistream header.
+
+The header refers to the initial message sent by a peer to initiate the protocol
+negotiation process. This message includes the protocol ID, `/multistream/1.0.0.`,
+which indicates to the listening peer that a protocol negotiation process is about
+to take place. Upon receiving this header, the listening peer responds with their
+desired protocol header.
```

Comment:
I wouldn't put too much emphasis on the header. It's not terribly important. Just say: Every multistream message is prefixed with a header, `/multistream/1.0.0`.

Reply:
Simplified